test/Dialect/FIRRTL/ref.mlir

```diff
@@ -209,17 +209,17 @@ firrtl.circuit "Forceable" {
     in %reset : !firrtl.uint<1>,
     in %value : !firrtl.uint<2>,
     out %node_ref : !firrtl.rwprobe<uint<2>>,
-    out %wire_ref : !firrtl.rwprobe<uint<2>>,
+    out %wire_ref : !firrtl.rwprobe<uint>,
     out %reg_ref : !firrtl.rwprobe<uint<2>>,
     out %regreset_ref : !firrtl.rwprobe<uint<2>>) {
 
     %n, %n_f = firrtl.node %value forceable : !firrtl.uint<2>
     firrtl.ref.define %node_ref, %n_f : !firrtl.rwprobe<uint<2>>
 
     // TODO: infer ref result existence + type based on "forceable" or other ref-kind(s) indicator.
-    %w, %w_f = firrtl.wire forceable : !firrtl.uint<2>, !firrtl.rwprobe<uint<2>>
-    firrtl.ref.define %wire_ref, %w_f : !firrtl.rwprobe<uint<2>>
-    firrtl.strictconnect %w, %value : !firrtl.uint<2>
+    %w, %w_f = firrtl.wire forceable : !firrtl.uint, !firrtl.rwprobe<uint>
```

Comment:
An infer-widths test with this would be awesome.

Reply:
Absolutely! I think the fix included gets that working, happily enough!
Other passes likely will need follow-on PR's, plan is to address enough here to not be broken and then split pass-specific updates into subsequent updates for sanity/review-friendliness :+1: .

Reply:
(Done!)